modules/calib3d/src/ptsetreg.cpp

```diff
@@ -501,6 +501,242 @@ class Affine3DEstimatorCallback : public PointSetRegistrator::Callback
     }
 };
 
+class Affine2DEstimatorCallback : public PointSetRegistrator::Callback
+{
+public:
+    int runKernel( InputArray _m1, InputArray _m2, OutputArray _model ) const
+    {
+        Mat m1 = _m1.getMat(), m2 = _m2.getMat();
+        const Point2f* from = m1.ptr<Point2f>();
+        const Point2f* to   = m2.ptr<Point2f>();
+
+        // we need 3 points to estimate affine transform
+        const int N = 6;
+        double buf[N*N + N + N];
+        Mat A(N, N, CV_64F, &buf[0]);
+        Mat B(N, 1, CV_64F, &buf[0] + N*N);
+        Mat X(N, 1, CV_64F, &buf[0] + N*N + N);
+        double* Adata = A.ptr<double>();
+        double* Bdata = B.ptr<double>();
+        A = Scalar::all(0);
+
+        for( int i = 0; i < (N/2); i++ )
+        {
+            Bdata[i*2] = to[i].x;
+            Bdata[i*2+1] = to[i].y;
+
+            double *aptr = Adata + i*2*N;
+            for(int k = 0; k < 2; ++k)
+            {
+                aptr[0] = from[i].x;
+                aptr[1] = from[i].y;
+                aptr[2] = 1.0;
+                aptr += N+3;
+            }
+        }
+
+        solve(A, B, X, DECOMP_SVD);
```

Comment:
looks nice, I will experiment with that tomorrow. (I'm reworking estimateAffine* functions API right now.)